src/modules/esl-utils/dom/events/listener.ts

```diff
@@ -0,0 +1,148 @@
+    if (typeof desc === 'string') return this.event === desc;
+    if (typeof desc === 'function') return this.handler === desc;
+    if (typeof desc === 'object') return isSimilar(this, desc);
+    return false;
+  }
+
+  /** Handles caught event (used as callback for low-level subscriptions) */
+  protected handle(e: Event): void {
+    if (this.isDelegatedTarget(e)) this.handler.call(this.$host, e, this);
+  }
+
+  /** Checks if the passed event can be handled by the curren event listener */
+  protected isDelegatedTarget(e: Event): boolean {
+    if (typeof this.selector !== 'string') return true;
+    const target = e.target;
+    const current = e.currentTarget;
+    if (!(target instanceof HTMLElement) || !(current instanceof HTMLElement)) return false;
+    return current.contains(target.closest(this.selector));
+  }
+
+  /** (Re-)Subscribes event listener instance */
+  public subscribe(): void {
+    this.unsubscribe();
+    memoize.clear(this, '$targets');
+    this.$targets.forEach((el: HTMLElement) => el.addEventListener(this.event, this.handle, this.capture));
+    ESLEventListener.get(this.$host).push(this);
+  }
+
+  /** Unsubscribes event listener instance */
+  public unsubscribe(): void {
+    this.$targets.forEach((el: HTMLElement) => el.removeEventListener(this.event, this.handle, this.capture));
+    const listeners = ESLEventListener.get(this.$host);
+    const value = listeners.filter((listener) => listener !== this);
+    Object.defineProperty(this.$host, STORE, {value, configurable: true});
+  }
+
+  /** Gets stored listeners array the passed `host` object */
+  public static get(host?: any): ESLEventListener[] {
+    if (!host) return [];
+    if (!Object.hasOwnProperty.call(host, STORE)) host[STORE] = [];
+    return host[STORE];
+  }
+
+  /** Gets descriptors from the passed object */
+  public static descriptors(target?: any): ESLListenerDescriptorFn[] {
+    if (!target) return [];
+    const desc: ESLListenerDescriptorFn[] = [];
+    for (const key in target) {
+      if (isDescriptorFn(target[key])) desc.push(target[key]);
+    }
+    return desc;
+  }
+
+  /** Creates event listeners by handler and descriptors */
+  public static create(target: HTMLElement, handler: ESLListenerHandler, desc: string | ESLListenerDescriptor): ESLEventListener[] {
+    desc = typeof desc === 'string' ? {event: desc} : desc;
+    return ESLEventListener.splitEventQ(desc.event).map((event) => {
+      const spec: ESLListenerDescriptor = Object.assign({}, desc, {event});
+      return new ESLEventListener(target, handler, spec);
+    });
+  }
+
+  /** Splits and deduplicate event query */
+  protected static splitEventQ(query: string): string[] {
+    const terms = (query || '').split(' ').map((term) => term.trim());
+    const deduplicate = new Set<string>();
+    return terms.filter((term) => {
+      if (!term || deduplicate.has(term)) return false;
+      deduplicate.add(term);
+      return true;
+    });
```

Comment:
Aggain, here we have a library core speciffic,  we should avoid slower operations.
Also we have a ES6 transpile specific here so in case we will have ES5 output it can appear to be even slower.

https://jsben.ch/4VNZG